element-hq/element-x-android
# Feature : Draft support

## Type of change


## Content
Implement draft support in the following way : 
- When message screen is backgrounded, then the current draft is saved. It also keeps the composer mode (either Regular, Reply or Edit).
- When message screen is first composed we load the draft and delete it from the store. 

This PR also introduce some refactoring around ReplyTo, so we can use the same UI logic between Timeline item and Composer. 



## Motivation and context
Closes https://github.com/element-hq/element-x-android/issues/2869


## Screenshots / GIFs



## Tests
- Open a room
- Write a message without sending it
- Close the room
- Open back the room 
- You should see your message in the composer
- You can repeat the steps with text formatting enabled, and reply/edit modes.


- Step 1
- Step 2
- Step ...

## Tested devices

- OS version(s):

## Checklist
